proposed/2021/ShowVulnerabilitiesInThePMUI.md

```diff
@@ -0,0 +1,90 @@
+* Qualitative feedback from customer interviews following the release of this feature show that customers find this feature very helpful for securing their NuGet packages.
+
+## Audience: Who are we building for?
+
+We are building this for Visual Studio users who manage and install packages through the package manager UI.
+
+## Scope
+
+**The MVP of this feature will be focused on enabling the following scenarios for Visual Studio .NET developers:**
+
+- Users can discover when they have top level packages that have a known vulnerabilities in the PMUI.
+- Users can see when a package search result in the Browse tab has a known vulnerability, to help them make informed decisions when choosing packages.
+- Users can find which installed top level packages have a vulnerability, to help them take the appropriate action.
+- Minor changes to the deprecation UX to align with necessary vulnerability UX.
+
+The following scenarios will not be in the MVP, but will be considered for next iterations:
+
+- Users can discover if they have installed packages with vulnerabilities through indicators in the solution explorer.
+- Users can see which transitively installed packages have vulnerabilities.
+- Users have a 1-click or wizard experience to “fix” vulnerabilities.
+- Plumbing deprecation and vulnerability data into the search API for a performance improvement.
+- Users must explicitly confirm that they would like to install a deprecated or vulnerable package version.
+- Users can choose to sort their installed packages such that deprecated or vulnerable packages rise to the top.
+- All deprecation UX changes requested in Home/VSdeprecationfeature.md at dev · NuGet/Home (github.com).
+
+The following scenarios are considered separate from this feature but may considered in another spec:
+- Users can discover if they have installed packages with vulnerabilities through warnings in restore.
+- Users can discover if they have installed packages with vulnerabilities through warning messages or dialogues outside of the PMUI.
+
+## MVP Design
+
+- Like deprecation, a warning sign will appear on the Installed tab header when a top-level deprecated package is installed.
+    - The warning sign on the Installed tab header will display the number of vulnerable and deprecated packages in the tooltip.
+The warning icon will appear for all levels of vulnerabilities – consistent with the behavior on NuGet.org.
+    - A bolded warning will be shown at the bottom on the package list item to make it more noticeable when a package is deprecated or vulnerable.
+- We will use the same warning sign for vulnerabilities as we do for deprecation to avoid the “lucky charms” effect where symbols get ignored because there are too many.
+- Deprecated and vulnerable packages will automatically be shown at the top of the installed package list in alphabetical order. The rest of the packages will be below in alphabetical order.
```

Comment:
Higher risk due the sorting's potential impact on other tabs / search results. Please move from MVP to future iteration.